api/src/main/java/com/github/retrooper/packetevents/protocol/world/chunk/palette/ListPalette.java

```diff
@@ -63,6 +64,12 @@ public ListPalette(int bitsPerEntry, PacketWrapper<?> wrapper) {
         this.nextId = paletteLength;
     }
 
+    public ListPalette(int bitsPerEntry, int[] data) {
+        this.bits = bitsPerEntry;
+        this.data = data;
```

Comment:
please verify the length of the data matches the bits-per-entry parameter

Reply:
Copying the array is fine :+1:
But if you do that, please always copy/clone it, so the constructor behaves consistently

Reply:
It's the behavior now.